csharp/ql/lib/semmle/code/csharp/frameworks/EntityFramework.qll

```diff
@@ -351,67 +352,137 @@ module EntityFramework {
 
     /** Holds if component stack `head :: tail` is required for the output specification. */
     predicate requiresComponentStackOut(
-      Content head, Type headType, SummaryComponentStack tail, int dist
+      Content head, Type headType, SummaryComponentStack tail, int dist,
+      DbContextClassSetProperty dbSetProp
     ) {
-      exists(Property dbSetProp, PropertyContent c1 |
+      exists(PropertyContent c1 |
         dbSetProp = this.getADbSetProperty(headType) and
         this.stepRev(c1, _, head, headType, 0) and
         c1.getProperty() = dbSetProp and
-        tail = SummaryComponentStack::jump(dbSetProp.getGetter()) and
+        tail = SummaryComponentStack::return() and
         dist = 0
       )
       or
       exists(Content tailHead, SummaryComponentStack tailTail, Type tailType |
-        this.requiresComponentStackOut(tailHead, tailType, tailTail, dist - 1) and
+        this.requiresComponentStackOut(tailHead, tailType, tailTail, dist - 1, dbSetProp) and
         tail = SummaryComponentStack::push(SummaryComponent::content(tailHead), tailTail) and
         this.stepRev(tailHead, tailType, head, headType, dist)
       )
     }
+
+    pragma[nomagic]
+    string getSyntheticNames(
+      SummaryComponentStack input, SummaryComponentStack output, DbContextClassSetProperty dbSet
+    ) {
+      exists(Property mapped |
+        this = dbSet.getDbContextClass() and
+        input(this, input, mapped) and
+        output(this, output, mapped, dbSet) and
+        result = dbSet.getSyntheticName() + "#" + SummaryComponentStack::getComponentStack(output)
+      )
+    }
   }
 
-  private class DbContextSaveChanges extends EFSummarizedCallable {
+  private class DbContextClassSetProperty extends Property {
     private DbContextClass c;
 
-    DbContextSaveChanges() { this = c.getASaveChanges() }
+    DbContextClassSetProperty() { this = c.getADbSetProperty(_) }
 
-    pragma[noinline]
-    private predicate input(SummaryComponentStack input, Property mapped) {
-      exists(PropertyContent head, SummaryComponentStack tail |
-        c.requiresComponentStackIn(head, _, tail, _) and
-        head.getProperty() = mapped and
-        mapped = c.getAColumnProperty(_) and
-        input = SummaryComponentStack::push(SummaryComponent::content(head), tail)
+    /**
+     * Gets the string representation for a synthetic identifier for this.
+     */
+    string getSyntheticName() {
```

Comment:
I would just call it `getQualifiedName` and update the QL doc.

Reply:
I don't think we can use that name as it still exists on `NamedElement` - will use `getFullName` instead.